### PR TITLE
Update documentation to reflect Apama plugin changes

### PR DESCRIPTION
**Pull request by jdoc-sag**
## Proposed changes

The Apama team are taking over responsibility for the Apama plugin.
This pull request updates the Apama plugin documentation following changes to the plugin made by the Apama team.

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)
- [x] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [ ] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: docs/src/howto-guides/017_apama_software_management_plugin.md [hunk 1]
```diff
@@ -1,70 +1,40 @@
...
 
-The apama plugin can be used to install and manage apama artefacts.
-This plugin can be used to install an apama project on a device as well as inject monitor files into an already installed project.
+The Apama plugin can be used to install Apama projects using the Cumulocity software management feature.
 
+> Note: This plugin expects an Apama installation on the device.
 
+## Install Apama artifacts from Cumulocity
 
-## Download and Install Apama Plugin
```
Comment (albinsuresh) [inline docs/src/howto-guides/017_apama_software_management_plugin.md:8]: Why did you remove this section on installing the Apama software management plugin itself? Is the plugin now part of the Apama installation itself? Otherwise, how to install it?

File: docs/src/howto-guides/017_apama_software_management_plugin.md [hunk 2]
```diff
@@ -1,70 +1,40 @@
...
-Add these binaries as software packages in Cumulocity software repository by following the instructions in the previous section.
-Once added, this apama project can be installed on any target device.
-You can test if the project got successfully installed by running the following Apama command:
+Add these binaries as software packages in the Cumulocity software repository by following the instructions in the previous section.
+Once added, this Apama project can be installed on any target device.
+You can test if the project was successfully installed by running the following Apama command:
 
 ```shell
-/opt/softwareag/Apama/bin/apama_env engine_inspect -m
+/opt/softwareag/Apama/bin/Apama_env engine_inspect -m
```
Comment (albinsuresh) [inline docs/src/howto-guides/017_apama_software_management_plugin.md:34]: Accidental side effect of "replace all"? Or has the binary name really changed?

File: docs/src/howto-guides/017_apama_software_management_plugin.md [hunk 3]
```diff
@@ -1,70 +1,40 @@
...
-1. Name must be suffixed with `::project` as in `my-demo-project::project`
-2. Version must be suffixed with `::apama` as in `1.0::apama` or just `::apama` if you don't need a version number.
-3. The uploaded binary must be a `zip` file that contains the `project` directory. If a directory named `project` is not found at the root level in the zip, it's considered invalid.
+1. The name must be suffixed with `::project` as in `my-demo-project::project`.
+2. The version must be suffixed with `::Apama` as in `1.0::Apama` or just `::Apama` if no  version number is necessary.
+3. The uploaded binary must be a `zip` file that contains the `project` directory. If a directory named `project` is not found by the plugin at the root level in the zip, it is considered invalid.
 
 ![Add new apama project in Software Repository](./images/apama-plugin/apama-project-c8y-software-repository.png)
 
-For apama monitor file:
```
Comment (albinsuresh) [inline docs/src/howto-guides/017_apama_software_management_plugin.md:42]: Why was this section on injecting `mon` files removed? Doesn't the new version of the plugin support this anymore?

Comment (jdoc-sag) [reply docs/src/howto-guides/017_apama_software_management_plugin.md:42]: Yes, the new version will not support this.

File: docs/src/howto-guides/017_apama_software_management_plugin.md (hunk 1 above)
Comment (jdoc-sag) [reply docs/src/howto-guides/017_apama_software_management_plugin.md:8]: The method of installation will change and I do not know what to put here yet. Some details have yet to be worked out, hence this is still a draft pull request.

Comment (jdoc-sag) [reply docs/src/howto-guides/017_apama_software_management_plugin.md:8]: 91ba8fb

File: docs/src/howto-guides/017_apama_software_management_plugin.md [hunk 4]
```diff
@@ -1,70 +1,59 @@
...
+## Add the Apama thin-edge support package to the tenant software repository and deploy it to thin-edge
 
+1. In the **Device Management** app and go to the **Management** menu option and select the **Software repository**.
+2. Click **Add software** at the right of the top menu bar.
+3. In the **ADD SOFTWARE** dialog enter the following details:
+    - **Software**: apama-thin-edge-support
+    - **Description**: apama-thin-edge-support (or something else if you choose)
+    - **Device Filter Type**: (leave empty)
+    - **Software Type**: apt
+    - **Version**: 10.15.1.0-1::apt
```
Comment (mjj29) [inline docs/src/howto-guides/017_apama_software_management_plugin.md:14]: This is the version from the 10.15.1 branch - it should be 10.15.0.2 in the published version

**Review by mjj29 — CHANGES_REQUESTED** [review]

File: docs/src/howto-guides/017_apama_software_management_plugin.md (hunk 4 above)
Comment (jdoc-sag) [reply docs/src/howto-guides/017_apama_software_management_plugin.md]: cc887df

**Review by mjj29 — APPROVED** [review]

File: docs/src/howto-guides/017_apama_software_management_plugin.md [hunk 5]
```diff
@@ -1,70 +1,59 @@
...
-Before an apama project or a mon files can be installed on the device using the software management feature in Cumulocity, these project files or monitor files need to be added to the Cumulocity software repository first.
-
-There's some naming convention that you need to follow while creating software entries for apama artefacts in the software repository.
-
-For apama projects:
-
-1. Name must be suffixed with `::project` as in `my-demo-project::project`
-2. Version must be suffixed with `::apama` as in `1.0::apama` or just `::apama` if you don't need a version number.
-3. The uploaded binary must be a `zip` file that contains the `project` directory. If a directory named `project` is not found at the root level in the zip, it's considered invalid.
+1. The name must be suffixed with `::project` as in `my-demo-project::project`.
```
Comment (albinsuresh) [inline docs/src/howto-guides/017_apama_software_management_plugin.md:34]: This `::project` suffix and `::mon` suffix were needed only when the plugin was supporting the installation of both projects and mon files. But, now that the plugin only supports project files **only**, you can avoid this suffix requirement.

Comment (jdoc-sag) [reply docs/src/howto-guides/017_apama_software_management_plugin.md]: cb58271

File: docs/src/howto-guides/017_apama_software_management_plugin.md [hunk 6]
```diff
@@ -1,70 +1,58 @@
...
+7. In the sub-menu for the device, select the **Software** option.
+8. Click the **Install software** button in the bottom left; the apama-thin-edge-support should be listed.
+9. Click the drop-down arrow on the right and check the 10.15.1.0-1::apama radio button, then click **Install**.
+10. Finally, click the **Apply changes** button in the lower right of the panel.
 
-## Download and Install Apama Plugin
+After installation of the support for Apama onto the thin-edge device, the correlator will attempt to start, but will fail initially because there is not yet an Apama project installed. Instructions are given below for installing an Apama project on the device.
 
-> Note: This plugin expects an apama installation on the device which is managed as a System V service. If Apama is not installed or if the System V init script is not installed, follow the Apama installation instructions [here](https://github.com/thin-edge/thin-edge.io_examples/tree/main/StreamingAnalytics#setup-and-configuration) before installing/using the apama plugin.
+## Install Apama artifacts from Cumulocity
```
Comment (albinsuresh) [inline docs/src/howto-guides/017_apama_software_management_plugin.md:26]:
```suggestion
## Install Apama projects from Cumulocity
```

File: docs/src/howto-guides/017_apama_software_management_plugin.md [hunk 7]
```diff
@@ -1,70 +1,58 @@
 # Apama Software Management Plugin
 
-The apama plugin can be used to install and manage apama artefacts.
-This plugin can be used to install an apama project on a device as well as inject monitor files into an already installed project.
+The Apama plugin, part of the Apama thin-edge support package, can be used to install Apama projects using the Cumulocity software management feature.
 
+## Add the Apama thin-edge support package to the tenant software repository and deploy it to thin-edge
```
Comment (albinsuresh) [inline docs/src/howto-guides/017_apama_software_management_plugin.md:5]:
```suggestion
## Prerequisites

The Apama thin-edge support package must be installed before you can install Apama projects on thin-edge.
If not already installed, it can also be installed from the Cumulocity software repository as follows:
```

File: docs/src/howto-guides/017_apama_software_management_plugin.md [hunk 8]
```diff
@@ -1,70 +1,58 @@
...
-Before an apama project or a mon files can be installed on the device using the software management feature in Cumulocity, these project files or monitor files need to be added to the Cumulocity software repository first.
-
-There's some naming convention that you need to follow while creating software entries for apama artefacts in the software repository.
-
-For apama projects:
-
-1. Name must be suffixed with `::project` as in `my-demo-project::project`
-2. Version must be suffixed with `::apama` as in `1.0::apama` or just `::apama` if you don't need a version number.
-3. The uploaded binary must be a `zip` file that contains the `project` directory. If a directory named `project` is not found at the root level in the zip, it's considered invalid.
+1. The version must be suffixed with `::Apama` as in `1.0::Apama` or just `::Apama` if no  version number is necessary.
```
Comment (albinsuresh) [inline docs/src/howto-guides/017_apama_software_management_plugin.md:34]: The earlier version of the `tedge-apama-plugin` that thin-edge team developed was installed at `/etc/tedge/sm-plugins` as `apama` instead of `Apama`. So, if you'd like to replace the existing installations of the old plugin with the new one, you may also wanna name your plugin binary as `apama` and that same name needs to be used as the suffix here. Otherwise both will coexist on machines where the older plugin is already installed. But, if the intent was to keep both side-by-side, then this is fine.

File: docs/src/howto-guides/017_apama_software_management_plugin.md (hunk 7 above)
Comment (albinsuresh) [inline docs/src/howto-guides/017_apama_software_management_plugin.md:5]: You might also wanna provide a link to where the user can download this apama thin-edge support package. I also hope that this support package will work even on a machine where apama is already installed, in which case, it will just install the Apama thin-edge plugin only.

File: docs/src/howto-guides/017_apama_software_management_plugin.md (hunk 8 above)
Comment (mjj29) [reply docs/src/howto-guides/017_apama_software_management_plugin.md:37]: Pretty sure we are using lowercase and this is just a doc issue. The package will force-remove the tedge-apama-plugin when it installs

File: docs/src/howto-guides/017_apama_software_management_plugin.md (hunk 7 above)
Comment (mjj29) [reply docs/src/howto-guides/017_apama_software_management_plugin.md]: There's not a fixed URL for the package - it depends on the version. The install instructions should be to install the apama-repo package (for which there is a fixed URL) and then install it as here using apt from the repo

File: docs/src/howto-guides/017_apama_software_management_plugin.md (hunk 8 above)
Comment (jdoc-sag) [reply docs/src/howto-guides/017_apama_software_management_plugin.md:37]: The file is lower case. This is a docs error caused by VS code, which I will fix.

Comment (jdoc-sag) [reply docs/src/howto-guides/017_apama_software_management_plugin.md]: a8d5ca5

File: docs/src/howto-guides/017_apama_software_management_plugin.md (hunk 7 above)
Comment (jdoc-sag) [reply docs/src/howto-guides/017_apama_software_management_plugin.md]: c622e75

File: docs/src/howto-guides/017_apama_software_management_plugin.md [hunk 9]
```diff
@@ -1,70 +1,82 @@
...
+#### Set up the repository from which the packages will be installed
+1. In the Cumulocity IoT tenant, open the **Device Management** app, go to the **Management** menu option and select the **Software repository**.
+2. Click **Add software** at the right of the top menu bar. 
+3. In the **ADD SOFTWARE** dialog enter the following details:
+- **Software**: apama-repo
+- **Description**: apama-repo (or something else if you choose)
+- **Device Filter Type**: (leave empty)
+- **Software Type**: apt
+- **Version**: 2022::apt
+- **SOFTWARE FILE**: check the Provide a file path option and enter the URL:	https://downloads.apamacommunity.com/debian/apama-repo_2022_all.deb
```
Comment (albinsuresh) [inline docs/src/howto-guides/017_apama_software_management_plugin.md:20]:
```suggestion
- **SOFTWARE FILE**: check the *Provide a file path* option and enter the URL:	https://downloads.apamacommunity.com/debian/apama-repo_2022_all.deb
```

File: docs/src/howto-guides/017_apama_software_management_plugin.md [hunk 10]
```diff
@@ -1,70 +1,82 @@
...
-
-1. Name must be suffixed with `::project` as in `my-demo-project::project`
-2. Version must be suffixed with `::apama` as in `1.0::apama` or just `::apama` if you don't need a version number.
-3. The uploaded binary must be a `zip` file that contains the `project` directory. If a directory named `project` is not found at the root level in the zip, it's considered invalid.
+The Apama plugin, part of the Apama thin-edge support package, can be used to install Apama projects using the Cumulocity software management feature.
+
+## Prerequisites
+
+The Apama thin-edge support package must be installed before you can install Apama projects on thin-edge.
+The recommended way of installing Apama support is with the Debian based distributions apt package manager.
```
Comment (albinsuresh) [inline docs/src/howto-guides/017_apama_software_management_plugin.md:8]:
```suggestion
The recommended way of installing Apama support on Debian based distributions is with the apt package manager.
```

File: docs/src/howto-guides/017_apama_software_management_plugin.md [hunk 11]
```diff
@@ -1,70 +1,82 @@
...
+Further details about this can be found in the thin-edge.io documentation at [Manage the software on your devices from Cumulocity cloud](https://thin-edge.github.io/thin-edge.io/html/tutorials/software-management.html).
+
+#### Set up the repository from which the packages will be installed
```
Comment (albinsuresh) [inline docs/src/howto-guides/017_apama_software_management_plugin.md:11]:
```suggestion
#### Set up the Debian repository from which the Apama packages will be installed

```

**Review by albinsuresh — APPROVED** [review]

**Comment by didier-wenzek** [comment]
Thank you for these updates
